unit-tests/rsutils/number/stabilized/test-stabilized.py

```diff
@@ -0,0 +1,220 @@
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "not full history" )
+#
+# Verify if history is not the logic works as expected and the percentage is calculated
+# from the history current size
+#
+try:
+    sv = stabilized_value( 30 )
+    sv.add( 76. )
+    sv.add( 76. )
+    sv.add( 76. )
+    sv.add( 76. )
+    test.check_equal( sv.get( 0.6 ), 76. )
+
+    sv.add( 45. )
+    sv.add( 45. )
+    sv.add( 45. )
+    sv.add( 45. )
+    sv.add( 45. )
+
+    test.check_equal( sv.get( 0.6 ), 76. )
+    sv.add( 45. )  # The stable value should change now (4 * 76.0 + 6 * 45.0 (total 10 values))
+    test.check_equal( sv.get( 0.6 ), 45. );
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "stable value sanity" )
+#
+# Verify if history is full with a specific value, the stabilized value is always the same
+# no matter what percentage is required
+#
+try:
+    sv = stabilized_value( 5 )
+    sv.add( 1. )
+    sv.add( 1. )
+    sv.add( 1. )
+    sv.add( 1. )
+    sv.add( 1. )
+
+    test.check_equal( sv.get( 1.   ), 1. )
+    test.check_equal( sv.get( 0.4  ), 1. )
+    test.check_equal( sv.get( 0.25 ), 1. )
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "stay with last value" )
+#
+# Verify if history is filled with less stable percantage than required the last stable
+# value is returned
+#
+try:
+    sv = stabilized_value( 10 )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+
+    test.check_equal( sv.get( 0.6 ), 55. )
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "update stable value - nominal" )
+#
+# Verify if history is filled with a stable value and then filled with required percentage
+# of new val, new val is returned as stable value.
+#
+try:
+    sv = stabilized_value( 10 )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 55. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+    sv.add( 60. )
+
+    test.check_equal( sv.get( 0.6 ), 60. )
+    sv.add( 35. )
+    sv.add( 35. )
+    sv.add( 35. )
+    sv.add( 35. )
+    sv.add( 35. )
+    test.check_equal( sv.get( 0.6 ), 60. )
+
+    sv.add( 35. )
+    test.check_equal( sv.get( 0.6 ), 35. )
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+test.print_results_and_exit()
```

Comment:
IMO 2 tests are missing:
```
TEST_CASE("update stable value - last stable not in history", "[stabilized value]")
TEST_CASE("update stable value - last stable is in history", "[stabilized value]")


```

Reply:
I didn't see them in the cpp files... you see them?

Reply:
Oh I think ,aybe you\'re right, I'll add

Reply:
Done